src/hooks/use-redirect-when-logged.ts

```diff
@@ -0,0 +1,21 @@
+import { useSelector } from "react-redux";
+import { AuthState } from "../features/auth/store/types";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
+
+function useAuthState(): AuthState {
+  return useSelector<{ auth: AuthState }>((state) => state.auth) as AuthState;
+}
+
+export default function useRedirectWhenLogged() {
+  const { isLogged, user } = useAuthState();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isLogged) {
+      navigate(`/dashboard`);
```

Comment:
Esto habria que hacerlo despues, no ahorita, pero yo creo que por defecto el redirect when logged seria a este lugar:
```suggestion
      navigate(`/profile/${....}`);
```